src/v2/Components/CookieConsentManager/CookieConsentManager.tsx

```diff
@@ -0,0 +1,299 @@
+import React, { useEffect, useState } from "react"
+import ConsentManagerBuilder from "@segment/consent-manager/commonjs/consent-manager-builder"
+// import inRegions from "@segment/in-regions"
+import { getENV } from "v2/Utils/getENV"
+import {
+  Box,
+  Button,
+  Checkbox,
+  Clickable,
+  Column,
+  DROP_SHADOW,
+  Flex,
+  GridColumns,
+  ModalDialog,
+  Spacer,
+  StackableBorderBox,
+  Text,
+} from "@artsy/palette"
+import { isEmpty } from "lodash"
+
+export const CookieConsentManager: React.FC = () => {
+  return (
+    <ConsentManagerBuilder writeKey={getENV("SEGMENT_WRITE_KEY")}>
+      {props => {
+        return <ConsentManagerModal {...props} />
+      }}
+    </ConsentManagerBuilder>
+  )
+}
+
+const ConsentManagerModal = ({
+  destinations,
+  preferences,
+  destinationPreferences,
+  setPreferences,
+  saveConsent,
+}) => {
+  const [showPreferences, setShowPreferences] = useState(false)
+  const categoryTable = getCategoryTable(destinations)
+
+  const handleSaveConsent = (...args) => {
+    saveConsent(...args)
+  }
+
+  useEffect(() => {
+    if (isEmpty(preferences)) {
+      return
+    }
+
+    const cookie = encodeURIComponent(
+      JSON.stringify({
+        version: 1,
+        destinations: destinationPreferences,
+        artsy: true,
+      })
+    )
+
+    fetch(`/set-tracking-preferences?trackingPreferences=${cookie}`)
```

Comment:
@dzucconi - this stuff has already been merged into server.tsx, hence the need for a rebase (and this being stripped out)